packages/data/src/components/with-select/with-select-old.js

```diff
@@ -0,0 +1,208 @@
+/**
```

Comment:
This file is just kept in this pull initially as a reference to compare the logic in the useSelect hook against what `withSelect` was doing.  Once an approach is signed off on, then this can get removed.